src/modules/krita/producer_krita.c

```diff
@@ -0,0 +1,224 @@
+/*
+ * filter_freeze.c -- simple frame freezing filter
+ * Copyright (C) 2022 Eoin O'Neill <eoinoneill1991@gmail.com>
+ * Copyright (C) 2022 Emmet O'Neill <emmetoneill.pdx@gmail.com>
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+#include <framework/mlt.h>
+#include <math.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <stdbool.h>
+#include <limits.h>
+
+#include <framework/mlt_frame.h>
+#include <framework/mlt_producer.h>
+#include <framework/mlt_service.h>
+#include <framework/mlt_factory.h>
+#include <framework/mlt_property.h>
+
+
+
+typedef struct
+{
+	int frame_start;
+	int frame_end;
+	int limit_enabled;
+	double speed;
+	mlt_producer producer_internal;
```

Comment:
Please format all files in the `krita` subfolder using `clang-format` 14 and our `.clang-format` file.